scribejava-apis/src/test/java/com/github/scribejava/apis/examples/GitHubAsyncOkHttpExample.java

```diff
@@ -0,0 +1,87 @@
+package com.github.scribejava.apis.examples;
+
+import com.github.scribejava.apis.GitHubApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthRequestAsync;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.httpclient.okhttp.OkHttpHttpClient;
+import okhttp3.OkHttpClient;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.Scanner;
+import java.util.concurrent.ExecutionException;
+
+public final class GitHubAsyncOkHttpExample {
+
+    private static final String NETWORK_NAME = "GitHub";
+    private static final String PROTECTED_RESOURCE_URL = "https://api.github.com/user";
+
+    private GitHubAsyncOkHttpExample() {
+    }
+
+    public static void main(String... args) throws IOException, ExecutionException, InterruptedException {
+        // Replace these with your client id and secret
+        final String clientId = "your client id";
+        final String clientSecret = "your client secret";
+        final String secretState = "secret" + new Random().nextInt(999_999);
+        final OAuth20Service service = new ServiceBuilder()
+                .apiKey(clientId)
+                .apiSecret(clientSecret)
+                .state(secretState)
+                .callback("http://www.example.com/oauth_callback/")
+                .httpClient(new OkHttpHttpClient(new OkHttpClient()))
```

Comment:
httpClient - is for externally created Http client.
It's easier to use httpClientConfig method